aips/aip-x.md

```diff
@@ -0,0 +1,105 @@
+
+If we delay enacting this, users will need to wait until their applications adopt allow-list frameworks to render content.
+
+### Out of Scope
+
+This is only a mitigation, a victim cannot opt-out of receiving unsolicited content nor automatically remove this content. Burning also does not result in the burner receiving a refund, but the cost of doing so is expected to be negligible, that is, the cost of transferring a digital asset.
+
+## Impact
+
+This will impact the following parties:
+* Users will be empowered to *burn* object-based assets from their account.
+* Smart contract developers will need to be cognizant that even *soul bound* objects can be moved to the burn address.
+* The primary fungible asset store for an account may accidentally be transferred away.
+* Wallet developers can enable access to a special function to *burn* object-based assets.
+* Indexers likely will need to know about the burn address to avoid indexing it as the amount of data there will likely be orders of magnitude greater than any other account.
+
+Note, as this enables primary fungible asset stores owner's to change, the primary fungible asset has access to a special `unburn` function that let's it reclaim ownership in the case that the original owner calls `primary_fungible_store::transfer` or `primary_fungible_store::withdraw`.
+
+## Alternative solutions
+
+1. Allow-lists
+  * (+) Allow-lists are the long term solution here as the blockchain grows in popularity, the number of scams and other unsolicited content will grow potentially faster than a user can manage their account. Furthermore, the costs may become prohibitive from cleaning up accounts.
+  * (-) Allow-lists require a lot of coordination and management, hence this is a longer term solution that will likely take large ecosystem support to setup.
+2. Deletion of user-owned objects
+  * (+) This allows a user to reclaim a refund for garbage data that likely exceeds the cost of performing the deletion, which in turn will reduce unnecessary storage load on the blockchain.
+  * (-) Some data structures cannot be deleted without potentially breaking applications on the blockchain, for example, a liquidity pool may be entirely broken if the fungible asset metadata were deleted. This could result in valuable assets being permanently locked on-chain without leveraging a governance proposal.
+
+The burn address solution was chosen as it provides a safe means to remove content associated with the users account without potentially breaking other applications.
+
+Note, the current proposal is for an `unburn` function to allow `primary_fungible_store` to reclaim accidentally transferred objects. This could instead be an alternative call into `primary_fungible_store` that explicitly has `unburn`, but that requires that wallets or other applications adopt the `unburn` functionality. The cost for checking the ownership should be negligible as the ownership must be checked prior to extracting assets.
```

Comment:
Are there going to be events emitted by burn/unburn?

Reply:
Hmm... without the new event harness, I'd be concerned about adding one-off events especially for super visible action such as burn. I do reckon ... unburn is quirky without events, so let's evaluate if we really want that and if we do, I'd probably make it defined on the signer for the burn address.

Reply:
Burn/Unburn can be introduced without events for now and we'll add events after event v2 is done